joblib/numpy_pickle.py

```diff
@@ -389,6 +393,8 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
         # By default, if compress is enabled, we want to be using 3 by
         # default
         compress = 3
+    if Path and isinstance(filename, Path):
```

Comment:
Nitpick: could you write this: "if Path is not None and isinstance(filename, Path)"